test/pummel/test-heapdump-env.js

```diff
@@ -16,8 +16,14 @@ const context = require('vm').createScript('const foo = 123');
 validateSnapshotNodes('Node / Environment', [{
   children: [
     { node_name: 'Node / cleanup_hooks', edge_name: 'cleanup_hooks' },
-    { node_name: 'process', edge_name: 'process_object' },
     { node_name: 'Node / IsolateData', edge_name: 'isolate_data' },
```

Comment:
This makes me wonder, do we still need this reference in Environment? I *think* that the compilers we support should be capable of optimizing out consecutive dereferences so that `env->principal_realm_->isolate_data_` isn't too different from `env_->isolate_data_`?

Reply:
Or I guess we can do it the other way around, adding a back reference to Environment in Realm, and `realm->isolate_data()` would be `realm->env_->isolate_data()` - I suppose realms would need to have access Environment eventually anyway, for states shared across realms (the isolate and the isolate data being part of those)

Reply:
We still have tons of access to isolate data properties with Environment. I think we should migrate these occurrence sites with realm properties before we move isolate_data off the Environment, i.e. removing [per-isolate property convenient accessors on Environment](https://github.com/nodejs/node/blob/main/src/env.h#L1336-L1338).

Reply:
hmm, but if the properties are per-isolate, why are we moving them off from the Environment? It doesn't make too much sense to have a copy of symbols and strings for each realm, either way, as they are shared by different contexts in the same isolate by nature.

Reply:
Maybe I misunderstand your suggestion. Do you suggest removing the isolate_data reference in realm or environment?

Reply:
I meant keeping it in Environment, and `realm->isolate_data()` is just `realm->env_->isolate_data()`, the compilers we support can probably optimize away the intermediate access.

Reply:
Thanks for the clarification, make sense to me. Updated.